### ISSUE TITLE
“Adoorable” achievement does not work

### ISSUE THREAD

**Issue by Wuzzy2** [kaadmy/pixture#24]
I can craft 1000 wooden doors and 1000000 stone doors and still don't get the “Adoorable” achievement.

Tested in: f01a0051c63bddae45d1b0ae7289a153a10c5f69

**Comment by kaadmy** [comment]
I kept renaming the door nodes, that'll probably fix it.

_Edit_: doesn't seem so, more testing..

**Comment by kaadmy** [comment]
Can you test with other achievements that also require crafting, like the Broadsword one?

**Comment by Wuzzy2** [comment]
I have successfully gotten all other crafting achievements.

EDIT: Oops! I just made a `git pull`. Big mistake. Now Pixture crashes when I open the achievements tab.

```
2016-11-14 00:59:21: ERROR[Main]: ServerError: Lua: Runtime error from mod 'default' in callback on_playerReceiveFields(): ...y/.minetest/games/pixture/mods/achievements/init.lua:152: attempt to index field '?' (a nil value)
2016-11-14 00:59:21: ERROR[Main]: stack traceback:
2016-11-14 00:59:21: ERROR[Main]:   ...y/.minetest/games/pixture/mods/achievements/init.lua:152: in function 'get_formspec'
2016-11-14 00:59:21: ERROR[Main]:   ...zy/.minetest/games/pixture/mods/default/formspec.lua:246: in function 'receive_fields'
2016-11-14 00:59:21: ERROR[Main]:   ...zy/.minetest/games/pixture/mods/default/formspec.lua:254: in function '?'
2016-11-14 00:59:21: ERROR[Main]:   /usr/share/minetest/builtin/game/register.lua:369: in function </usr/share/minetest/builtin/game/register.lua:349>
```

7f6b73246d7524d34f7c800edd51b6cef0cd227c

**Comment by asl97** [comment]
###### Cause:

https://github.com/kaadmy/pixture/blob/c8e0dbfe68fafea304b09d7fcfe0f781110670c4/mods/door/init.lua#L35

`register_craftitem` wasn't given the group definition.

###### Test method:

Adding a print after the register_craft

```
diff --git a/mods/door/init.lua b/mods/door/init.lua
index ed3003c..07639f1 100644
--- a/mods/door/init.lua
+++ b/mods/door/init.lua
@@ -292,6 +292,8 @@ minetest.register_craft(
       }
    })

+print(dump(ItemStack("door:door_wood"):get_definition()))
+
```

###### Test output:

As you can see, it is missing the groups

```
{
    inventory_image = "door_wood.png",
    mod_origin = "door",
    type = "craft",
    name = "door:door_wood",
    on_place = <function>,
    description = "Wooden Door"
}
```

###### Reason:

The door nodes was indeed given the door group.
However the item being crafted wasn't the door node but the door item which doesn't have the door group.

###### Solution:

Add `groups = def.groups,` into the `register_craftitem`

**Comment by asl97** [comment]
reason:
the door nodes was indeed given the door group,
however the item being crafted wasn't the door node but the door item which doesn't have the door group.

solution:
Add `groups = def.groups,` into the `register_craftitem`

**Comment by kaadmy** [comment]
I am so stupid, I checked this yesterday but didn't see that I called door.register_door and copied over attributes by hand.

**Comment by kaadmy** [comment]
FIxed in f38da4b
